### ISSUE TITLE
Project would benefit from an automated build

### ISSUE THREAD

**Issue by DavidWhitlock** [alexander94dmitriev/WebImageCollector#2]
Hey, @alexander94dmitriev.

It looks like you're making good progress on the project and I'd like to be able to build it and work with it, myself.  However, right now, I don't know how to do that.

I highly recommend that you create a Maven or Gradle project for the WebImageCollector.  That will enable the two of us to collaborate on the project more easily because we will be able to both build the project consistently.  (As I noted in #1, sharing IntelliJ files and binary artifacts built on each others' machines will not be very effective.)

It looks like there is a Maven archetype for PrimeFaces that should be a good starting place: http://www.techvalueinsight.com/blog/primefaces-setup-with-maven-and-tomcat

This will give us the ability to build the WebImageCollector in a consistent way.  (And, as you'll recall, "building" can include running automated tests!)

**Comment by alex94dmit** [comment]
Hey, @DavidWhitlock 

Thanks for the heads up! I wasn't sure what will be the appropriate way to build the project in this case. But looks like Maven is definetly the best solution in this case.
However, I do know how to make a Maven project and assign its own groupID(defined by me) with Primefaces artifact. bot how can I make sure that YOU will be able to build the project? How does the Maven works from the perspective of sharing source files?
When you say you want to make sure to work with the project you mean by running a jar file, just like we did on summer?

**Comment by DavidWhitlock** [comment]
I presume that if you put the project's source code in the src/main directory like Maven expects that I'll be able to build it.  We should test that out, though.

I don't think the application should have an executable jar file like the course's projects.  However, I would like to be able to use "mvn jetty:run-war" to run the application in a Jetty web container.  That's kind of like Projects 4 and 5, I guess.

Make sense?

**Comment by alex94dmit** [comment]
Got it!

Now I was able to resolve all dependencies and it should work pretty much fine with jetty. However, I was not able to configure my FileServlet, which should catch all responses from "/resources/images/" and transfer to "c:/images/". In the original project I was using Glassfish(if that's important), defined servlet name and mappings. Everything was working fine, but now it doesn't work. In the debugger mode I'm even never enter the class itself.

**Comment by alex94dmit** [comment]
I have uploaded the Maven project to repository. Give it a try!

**Comment by alex94dmit** [comment]
Nevermind, I believe I fixed it.
The Maven project seems to be working a bit different with a path, so I just should've account for that.

**Comment by DavidWhitlock** [comment]
Sweet.  I cloned your repository and then build the project in the WebImageCollectorJsf using "mvn clean verify" and then "mvn jetty:run-war".

While starting up Jetty warned that

```
[WARNING] Failed startup of context o.e.j.m.p.JettyWebAppContext@70e94ecb{/WebImageCollector,[file:///Users/david/PSU/git/alexander94dmitriev/WebImageCollector/WebImageCollectorJsf/target/WebImageCollector/, jar:file:///Users/david/PSU/git/alexander94dmitriev/WebImageCollector/WebImageCollectorJsf/target/WebImageCollector/WEB-INF/lib/primefaces-6.0.jar!/META-INF/resources, jar:file:///Users/david/PSU/git/alexander94dmitriev/WebImageCollector/WebImageCollectorJsf/target/WebImageCollector/WEB-INF/lib/jsf-impl-2.2.12.jar!/META-INF/resources],UNAVAILABLE}{/Users/david/PSU/git/alexander94dmitriev/WebImageCollector/WebImageCollectorJsf/target/WebImageCollector.war}
MultiException[javax.servlet.UnavailableException: main.FileServlet, javax.servlet.UnavailableException: main.FileServlet]
```

And when I visited http://localhost:9090/WebImageCollector/ in my browser, I got an error 503 "Service Unavailable" error.

It might be caused by this:

```
[WARNING] 
java.lang.ClassNotFoundException: main.FileServlet
```

**Comment by alex94dmit** [comment]
Hm, I was getting this error a while ago when I created Maven project. In the end after buld it's just complained about the naming convention of packages. So I fixed that.

If I clone it now, it works. Can you try making a clean cloning of repo again?

**Comment by alex94dmit** [comment]
Also, make sure to add "index.xhtml" to your path

As of running Jetty, "jetty:run" works fine for me, but with "jetty:run-war" I'm getting the same error as you.

**Comment by alex94dmit** [comment]
Aaaah, I got it.

FileServlet was outside of main/java folder, so when I opened web.xml it complained about the class not being found. It's weird that it worked before.

So now it should work. Please, refresh the repo.

**Comment by DavidWhitlock** [comment]
Cool.  I updated my checkout, rebuilt, and I saw this when I ran the web application.  Nice!

I recommend that you make index.xhtml the default welcome file: https://stackoverflow.com/questions/3976160/how-does-tomcat-find-the-home-page-of-my-web-app

<img width="1022" alt="screen shot 2017-10-09 at 5 24 11 am" src="https://user-images.githubusercontent.com/2935912/31337896-20968530-acb2-11e7-9888-efb0d68967bf.png">
